acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/privacy/multitenancy/OnChainMultiTenancyAcceptanceTest.java

```diff
@@ -0,0 +1,697 @@
+            multiTenancyPrivacyNode.getTokenForTenant(tenant));
+    final List<PrivacyRequestFactory.OnChainPrivacyGroup> group =
+        privacyNode.execute(
+            privacyTransactions.findOnChainPrivacyGroup(Base64String.unwrapList(base64StringList)));
+    assertThat(group.size()).isEqualTo(1);
+    assertThat(group.get(0).getMembers()).containsAll(base64StringList).hasSize(2);
+
+    // check that when you are not a member you cannot find the privacy group
+    privacyNode
+        .getBesu()
+        .useAuthenticationTokenInHeaderForJsonRpc(
+            multiTenancyPrivacyNode.getTokenForTenant(removedTenant));
+    assertThatThrownBy(
+            () ->
+                privacyNode.execute(
+                    privacyTransactions.findOnChainPrivacyGroup(
+                        Base64String.unwrapList(base64StringList))))
+        .hasMessageContaining("Error finding onchain privacy group");
+
+    // check that a member can do a priv_call
+    privacyNode
+        .getBesu()
+        .useAuthenticationTokenInHeaderForJsonRpc(
+            multiTenancyPrivacyNode.getTokenForTenant(tenant));
+    final EthCall readValue =
+        privacyNode.execute(
+            privacyTransactions.privCall(
+                privacyGroupId, eventEmitter, eventEmitter.value().encodeFunctionCall()));
+    assertThat(new BigInteger(readValue.getValue().substring(2), 16))
+        .isEqualByComparingTo(BigInteger.valueOf(VALUE_SET));
+
+    // check that when you are not a member you cannot do a priv_call
+    privacyNode
+        .getBesu()
+        .useAuthenticationTokenInHeaderForJsonRpc(
+            multiTenancyPrivacyNode.getTokenForTenant(removedTenant));
+    assertThatThrownBy(
+            () ->
+                privacyNode.execute(
+                    privacyTransactions.privCall(
+                        privacyGroupId, eventEmitter, eventEmitter.value().encodeFunctionCall())))
+        .hasMessageContaining("Unauthorized");
+
+    // check that a member can do a priv_getTransaction
+    privacyNode
+        .getBesu()
+        .useAuthenticationTokenInHeaderForJsonRpc(
+            multiTenancyPrivacyNode.getTokenForTenant(tenant));
+    final PrivacyRequestFactory.GetPrivateTransactionResponse privTransaction =
+        privacyNode.execute(privacyTransactions.privGetTransaction(storeTransactionHash));
+    assertThat(privTransaction.getResult().getPrivacyGroupId()).isEqualTo(privacyGroupId);
+
+    // check that when you are not a member you cannot do a priv_getTransaction
+    privacyNode
+        .getBesu()
+        .useAuthenticationTokenInHeaderForJsonRpc(
+            multiTenancyPrivacyNode.getTokenForTenant(removedTenant));
+    assertThatThrownBy(
+            () -> privacyNode.execute(privacyTransactions.privGetTransaction(storeTransactionHash)))
+        .hasMessageContaining(
+            "Expecting actual not to be null"); // TODO: returning null because the RPC is using the
```

Comment:
Is this a TODO or just an informative comment?

Reply:
This is informative. Just to show what the response is here when we are not part of the group!

Reply:
It is better to remove the `TODO` then :)

Reply:
Done